src/main/java/org/folio/service/util/EventHandlingUtil.java

```diff
@@ -1,14 +1,116 @@
 package org.folio.service.util;
 
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.Json;
+import io.vertx.kafka.client.producer.KafkaHeader;
+import io.vertx.kafka.client.producer.KafkaProducer;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.kafka.KafkaConfig;
+import org.folio.kafka.KafkaTopicNameHelper;
 import org.folio.processing.events.utils.PomReaderUtil;
+import org.folio.rest.jaxrs.model.Event;
+import org.folio.rest.jaxrs.model.EventMetadata;
 import org.folio.rest.tools.utils.ModuleName;
 
+import java.util.List;
+import java.util.UUID;
+
 public final class EventHandlingUtil {
 
+  private static final Logger LOGGER = LogManager.getLogger();
+
   private EventHandlingUtil(){}
 
   public static String constructModuleName() {
     return PomReaderUtil.INSTANCE.constructModuleVersionAndVersion(ModuleName.getModuleName(),
       ModuleName.getModuleVersion());
   }
+
+  /**
+   * Prepares and sends event with payload to kafka
+   *
+   * @param tenantId tenant id
+   * @param eventPayload eventPayload in String representation
+   * @param eventType eventType
+   * @param kafkaHeaders kafka headers
+   * @param kafkaConfig kafka config
+   * @return completed future with true if event was sent successfully
+   */
+  public static Future<Boolean> sendEventToKafka(
+      String tenantId,
+      String eventPayload,
+      String eventType,
+      List<KafkaHeader> kafkaHeaders,
+      KafkaConfig kafkaConfig,
+      String key,
+      Vertx vertx) {
+    LOGGER.debug("Starting to send event to Kafka for eventType: {}", eventType);
+    Event event = createEvent(eventPayload, eventType, tenantId);
+
+    String topicName = createTopicName(eventType, tenantId, kafkaConfig);
+
+    KafkaProducerRecord<String, String> record = createProducerRecord(event, key, topicName, kafkaHeaders);
+
+    Promise<Boolean> promise = Promise.promise();
+
+    String chunkId = extractHeader(kafkaHeaders, "chunkId");
+    String recordId = extractHeader(kafkaHeaders, "recordId");
+
+    String producerName = eventType + "_Producer";
+    KafkaProducer<String, String> producer =
+      KafkaProducer.createShared(vertx, producerName, kafkaConfig.getProducerProps());
+    producer.write(record, war -> {
+      producer.end(ear -> producer.close());
+      if (war.succeeded()) {
+        logSendingSucceeded(eventType, chunkId, recordId);
+        promise.complete(true);
+      } else {
+        Throwable cause = war.cause();
```

Comment:
We need to fail promise like this: promise.fail(cause);

Reply:
Agreed. This was modeled after code in SRM codebase.

Reply:
We could start thinking to moving common codebase for EventHandlingUtils to folio-kafka-wrapper to avoid code duplication, but  its not in scope of this PR of course

Reply:
Agreed as well. I wanted to do that but decided against it so that my first PR is not bloated.